### PR TITLE
Refactor api token based login to UAA, fix refresh token logic

### PR DESCRIPTION
**Pull request by vuil**
### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

Updated unit tests

Manual testing
```
Prior to fix, artificially roll back expiration time then run
> ./bin/tanzu_before_fix context get-token tspmapitoken
will trigger interactive login....

[i] Opening the browser window to complete the login
Log in by visiting this link:

    https://xxxxhostauth/oauth/authorize?client_id=tp_cli_app&code_challenge=3rgxYwz2lRK9ZK9kwhdJW85MW5cyutFdN54AZ-drtk0&code_challenge_method=S256&redirect_uri=http%3A%2F%2F127.0.0.1%3A60475%2Fcallback&response_type=code&state=c97c3bca05ff7145bad12013ccdbba91

After fix, the commands performs token refresh successfully
> ./bin/tanzu context get-token tspmapitoken
{"kind":"ExecCredential","apiVersion":"client.authent....

```

Tested login and context create with TANZU_API_TOKEN

```
 TANZU_API_TOKEN=3bf6c287c61243fcb826280b08dad345-r ./bin/tanzu context create tspmapitoken --type tanzu --endpoint https://xxxhost

 TANZU_API_TOKEN=3bf6c287c61243fcb826280b08dad345-r ./bin/tanzu login --endpoint https://xxxhost
```

### Release note

```release-note
Fix issue leading to premature triggering of interactive login.
```

### CODE REVIEW

**Review by marckhouzam — APPROVED** [review]
LGTM
Just one question

File: pkg/auth/common/login_handler.go [hunk 1]
```diff
@@ -166,18 +176,23 @@ func WithListenerPortFromEnv(envVarName string) LoginOption {
...
 
 func (h *TanzuLoginHandler) DoLogin() (*Token, error) {
+	var err error
+	var token *Token
+
 	if h.refreshToken != "" {
-		token, err := h.getTokenWithRefreshToken()
-		if err == nil {
-			return token, nil
+		ctx := contextWithCustomTLSConfig(context.TODO(), h.getTLSConfig())
```
[the remark this answers is not in the document]
Comment (vuil) [reply pkg/auth/common/login_handler.go:183]: makes sense, updating.

File: go.mod [hunk 2]
```diff
@@ -6,6 +6,8 @@ replace cloud.google.com/go => cloud.google.com/go v0.102.1
 
 replace github.com/vmware-tanzu/tanzu-cli/test/e2e/framework => ./test/e2e/framework
 
+replace golang.org/x/oauth2 => ../oauth2
```
Comment (prkalle) [inline go.mod]: question: is it intentional?

Comment (prkalle) [reply go.mod]: I see the PR is updated.

**Pull request merged**